examples/src/main/java/com/mapbox/navigation/examples/core/MapboxDropInActivity.kt

```diff
@@ -31,5 +43,58 @@ class MapboxDropInActivity : AppCompatActivity() {
         binding.tempStartNavigation.setOnClickListener {
             mapboxNavigationViewApi.temporaryStartNavigation()
         }
+
+        mapboxNavigationViewApi.getMapView()
+            .gestures
+            .addOnMapLongClickListener(onMapLongClickListener)
+    }
+
+    private val onMapLongClickListener = OnMapLongClickListener { point ->
+        getLastLocation(
+            this@MapboxDropInActivity,
+            WeakReference<(Expected<Exception, LocationEngineResult>) -> Unit> { result ->
+                result.fold(
+                    {
+                        Log.e(TAG, "Error obtaining current location", it)
+                    },
+                    {
+                        it.lastLocation?.let { currentLocation ->
+                            mapboxNavigationViewApi.fetchAndSetRoute(
+                                listOf(
+                                    Point.fromLngLat(
+                                        currentLocation.longitude,
+                                        currentLocation.latitude
+                                    ),
+                                    point
+                                )
+                            )
+                        }
+                    }
+                )
+            }
```

Comment:
That's a wrong way to use `WeakReference`. You create a weak reference to the lambda, but there are no strong references to it, so it is going to be removed during next GC cycle. You probably meant to create a weak reference for `mapboxNavigationViewApi` itself.